### PR TITLE
Added network check before attempting to install requirements

### PR DESCRIPTION
**Pull request by ima9rd**
Added network check before attempting to install requirements to prevent lengthy try/fail install process for each module in requirements.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -17,11 +21,25 @@ def install_and_import(package):
...
+# https://stackoverflow.com/questions/3764291/checking-network-connection
+def have_internet():
+    conn = httplib.HTTPConnection("www.google.com", timeout=5)
+    try:
+        conn.request("HEAD", "/")
+        conn.close()
+        return True
+    except:
+        conn.close()
+        return False
```
Comment (42B) [inline run.py]: Couldn't both these `conn.close()` be removed and a single call just be in a `finally` clause?

**Comment by tarehart** [comment]
This is working really well for me!

- For some reason, the print statement about NewConnnectionError was very delayed for me, e.g. after getting the DLL connection or closing the GUI. I was able to fix it by flushing: https://stackoverflow.com/questions/230751/how-to-flush-output-of-print-function
- Our users are easily startled by error messages, and seeing "NewConnectionError: unable to connect to install requirements" will probably make some folks arrive in issues-and-bugs. To avoid that, I would prefer a friendlier message like "Skipping upgrade check for now since it looks like you have no internet :)"
- It would be nice to put as much of this logic inside `rlbot` as possible. For example, we could make have_internet part of rlbot.utils.public_utils which allows us to
  - maintain the code in only one place, saving effort
  - send people updates to that code without requiring them to pull from git (or in many cases manually update their files since many of our users are uncomfortable with git)

Let's treat this as an excellent proof of concept, and proceed by refactoring as much logic into https://github.com/RLBot/RLBot/blob/master/src/main/python/rlbot/utils/public_utils.py as possible.

File: run.py [hunk 2]
```diff
@@ -17,11 +21,25 @@ def install_and_import(package):
...
+        conn.close()
+
+
 if __name__ == '__main__':
     install_and_import('rlbot')
     from rlbot.utils import public_utils
 
-    if public_utils.is_safe_to_upgrade():
+    if not have_internet():
+        print('NewConnectionError: unable to connect to install requirements.')
```
Comment (dtracers) [inline run.py]: we should use the rlbot logger

**Comment by dtracers** [comment]
We can use the RLBot Logger instead of print.
Though that does depend on the library existing at all

**Comment by ima9rd** [comment]
Good call, I have been familiarizing myself with the RLBot project all day. I will work on integrating this into public_util and rework to use the logger instead of print.

**Comment by dtracers** [comment]
We need to do a check then if we downloaded it at all

**Comment by ima9rd** [comment]
Updated to use the new have_network method from public_utils, as well as implemented RLBot logger in script. Unsure if logger was implemented correctly, so feedback is welcome.

File: run.py [hunk 3]
```diff
@@ -19,9 +21,12 @@ def install_and_import(package):
 
 if __name__ == '__main__':
     install_and_import('rlbot')
-    from rlbot.utils import public_utils
-
-    if public_utils.is_safe_to_upgrade():
+    from rlbot.utils import public_utils, logging_utils
+    import logging
```
Comment (dtracers) [inline run.py]: this is not needed if you are using the rlbot logging utils

Comment (ima9rd) [reply run.py]: Removed import for logging and used the RLBot log level instead

**Pull request merged**